### PR TITLE
remove submodule requirement for dependency fetching

### PR DESCRIPTION
**Pull request by chayim**

### CODE REVIEW

File: opt/build/libtorch/repack.sh [hunk 1]
```diff
@@ -24,8 +24,13 @@ else
...
 
-OS=$(python3 $ROOT/opt/readies/bin/platform --os)
-ARCH=$(python3 $ROOT/opt/readies/bin/platform --arch)
+# set them internally or externally
+if [ -z ${OS} ]; then
+    OS=$(python3 $ROOT/opt/readies/bin/platform --os)
+fi
+if [ -z ${ARCH} ]; then
+    ARCH=$(python3 $ROOT/opt/readies/bin/platform --arch)
+fi
```
Comment (DvirDukhan) [inline opt/build/libtorch/repack.sh:33]: part of readis repo, no?

File: get_deps.sh [hunk 2]
```diff
@@ -47,10 +47,17 @@ else
 	fi
 fi
 
-git submodule update --init --recursive || true
+#git submodule update --init --recursive || true
```
Comment (DvirDukhan) [inline get_deps.sh:50]: now that it is gone, where it should be called?

File: opt/build/libtorch/repack.sh (hunk 1 above)
Comment (chayim) [reply opt/build/libtorch/repack.sh:33]: So this allows us to use readies if it's there, if not we have an _approximation_ based on the outputs we received from the script. I'd say this isn't ideal - but it's a good enough first step.

File: get_deps.sh (hunk 2 above)
Comment (chayim) [reply get_deps.sh:50]: It shouldn't be. This is needed only for the content of the if statements. I'd argue if you clone without recursive, it needs to run, hence this change.

**Review by DvirDukhan — APPROVED** [review]

**Comment by DvirDukhan** [comment]
@Spartee please add your review

**Comment by Spartee** [comment]
I could not get this to work on MacOS

Steps I followed

1. `` git clone https://github.com/RedisAI/RedisAI.git --branch ck-optional-readies-depsonly --depth=1``

which seems to still download (or at least filter?) the 400Mb of git-lfs content

```
❯ git clone https://github.com/RedisAI/RedisAI.git --branch ck-optional-readies-depsonly --depth=1
Cloning into 'RedisAI'...
remote: Enumerating objects: 346, done.
remote: Counting objects: 100% (346/346), done.
remote: Compressing objects: 100% (321/321), done.
remote: Total 346 (delta 38), reused 194 (delta 9), pack-reused 0
Receiving objects: 100% (346/346), 4.50 MiB | 24.14 MiB/s, done.
Resolving deltas: 100% (38/38), done.
Filtering content: 100% (30/30), 395.78 MiB | 36.30 MiB/s, done.
```

2. ``./get_deps.sh``

which gives

```
x86_64
Cloning dlpack ...
Cloning into 'dlpack'...
remote: Enumerating objects: 36, done.
remote: Counting objects: 100% (36/36), done.
remote: Compressing objects: 100% (26/26), done.
remote: Total 36 (delta 0), reused 23 (delta 0), pack-reused 0
Unpacking objects: 100% (36/36), done.
Done.
Installing TensorFlow ...
There are errors.
```


the verbose output from ``VERBOSE=1 ./get_deps.sh``

```
❯ VERBOSE=1 ./get_deps.sh
+ [[ '' == \c\p\u ]]
+ [[ '' == 1 ]]
+ [[ '' == \g\p\u ]]
+ [[ '' == 1 ]]
+ GPU=0
+ [[ 0 == 1 ]]
+ DEVICE=cpu
+ '[' -f /Users/spartee/scratch/SmartSim/smartsim/.third-party/RedisAI/opt/readies/bin/platform ']'
++ uname -s
++ tr '[:upper:]' '[:lower:]'
+ OS=darwin
+ uname -m
+ grep aarch64
+ ARCH=x64
+ uname -m
+ grep x86
x86_64
+ [[ darwin == macos ]]
+ DEPS_DIR=/Users/spartee/scratch/SmartSim/smartsim/.third-party/RedisAI/deps/darwin-x64-cpu
+ mkdir -p /Users/spartee/scratch/SmartSim/smartsim/.third-party/RedisAI/deps/darwin-x64-cpu
+ cd /Users/spartee/scratch/SmartSim/smartsim/.third-party/RedisAI/deps/darwin-x64-cpu
+ DLPACK=dlpack
+ LIBTENSORFLOW=libtensorflow
+ LIBTFLITE=libtensorflow-lite
+ LIBTORCH=libtorch
+ MKL=mkl
+ ONNXRUNTIME=onnxruntime
+ DLPACK_VERSION=v0.5_RAI
+ [[ '' != 0 ]]
+ [[ '' == 1 ]]
+ [[ ! -d dlpack ]]
+ echo 'dlpack is in place.'
dlpack is in place.
+ TF_VERSION=2.4.0
+ [[ '' != 0 ]]
+ [[ '' == 1 ]]
+ [[ ! -d libtensorflow ]]
+ echo 'Installing TensorFlow ...'
Installing TensorFlow ...
+ [[ darwin == linux ]]
+ [[ darwin == macos ]]
+ LIBTF_ARCHIVE=libtensorflow----2.4.0.tar.gz
+ [[ ! -f libtensorflow----2.4.0.tar.gz ]]
+ wget -q /libtensorflow----2.4.0.tar.gz
++ error
++ echo 'There are errors.'
There are errors.
++ exit 1

```


If I was meant to still include the ``--recursive``, I tried that and I think that will work but still running into the ONNX on MacOS bug with the bad link which I think is documented in #743 


I can try this on other systems, but am I doing something wrong? Am I still supposed to use ``--recursive``? isn't the point of this PR to get rid of the submodules? apologies if I have missed something.

**Comment by chayim** [comment]
> 1. `git clone https://github.com/RedisAI/RedisAI.git --branch ck-optional-readies-depsonly --depth=1`

@Spartee Is this solved by the GIT_LFS_SKIP_SMUDGE=1 change that we discussed? When running  _GIT_LFS_SKIP_SMUDGE=1 git clone https://github.com/RedisAI/RedisAI.git --branch ck-optional-readies-depsonly --depth=1_  I receive the following output:

```
Cloning into 'RedisAI'...
remote: Enumerating objects: 346, done.
remote: Counting objects: 100% (346/346), done.
remote: Compressing objects: 100% (321/321), done.
remote: Total 346 (delta 38), reused 194 (delta 9), pack-reused 0
Receiving objects: 100% (346/346), 4.50 MiB | 4.36 MiB/s, done.
Resolving deltas: 100% (38/38), done.
```

Given that delta, I think this can now merge.  The point of the PR isn't to remove submodules - only to disable updating those submodules as part of our get_deps run.

**Pull request merged**